api/api_test.go

```diff
@@ -0,0 +1,623 @@
+package api_test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/eth/filters"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/onflow/flow-evm-gateway/api"
+	"github.com/onflow/flow-evm-gateway/storage"
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type Account struct {
+	key  *ecdsa.PrivateKey
+	addr common.Address
+}
+
+func TestBlockChainAPI(t *testing.T) {
+
+	t.Parallel()
+
+	store := storage.NewStore()
+	blockchainAPI := &api.BlockChainAPI{
+		Store: store,
+	}
+
+	t.Run("ChainId", func(t *testing.T) {
+		chainID := blockchainAPI.ChainId()
+
+		assert.Equal(t, (*hexutil.Big)(emulator.FlowEVMTestnetChainID), chainID)
+	})
+
+	t.Run("BlockNumber", func(t *testing.T) {
+		blockNumber := blockchainAPI.BlockNumber()
+
+		assert.Equal(t, hexutil.Uint64(0), blockNumber)
+	})
+
+	t.Run("Syncing", func(t *testing.T) {
+		syncing, err := blockchainAPI.Syncing()
+		require.NoError(t, err)
+
+		isSyncing := syncing.(bool)
+		assert.False(t, isSyncing)
+	})
+
+	t.Run("SendRawTransaction", func(t *testing.T) {
+		hash, err := blockchainAPI.SendRawTransaction(
+			context.Background(),
+			hexutil.Bytes{},
+		)
+		require.NoError(t, err)
+
+		assert.Equal(
+			t,
+			common.HexToHash("0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"),
+			hash,
+		)
+	})
+
+	t.Run("CreateAccessList", func(t *testing.T) {
```

Comment:
What I mentioned above about having a indexer (or multiple modules) defined as interface and then used in the API, would also allow for better testing, as you would use a test mock (maybe using tools like mockery) with which you can then specify the values and assert the returned values. This way, like you made it now, it will soon break all the tests as you heavily rely on the mocked values in the API itself, which you will soon start replacing with real values.

Reply:
Agreed, this was mainly for my convenience, to see what kind of return values these methods require, and to possibly use these later on as fixtures.